cinder/backup/manager.py

```diff
@@ -498,12 +482,65 @@ def _run_backup(self, context, backup, volume):
                                     backup_device.device_obj, properties,
                                     backup_device.is_snapshot, force=True,
                                     ignore_errors=True)
+        except Exception as err:
+            with excutils.save_and_reraise_exception():
+                if snapshot_id:
+                    snapshot.status = fields.SnapshotStatus.AVAILABLE
+                    snapshot.save()
+                else:
+                    self.db.volume_update(
+                        context, volume_id,
+                        {'status': previous_status,
+                         'previous_status': 'error_backing-up'})
+                self._update_backup_error(backup, six.text_type(err))
         finally:
             with backup.as_read_deleted():
                 backup.refresh()
             self._cleanup_temp_volumes_snapshots_when_backup_created(
                 context, backup)
-        return updates
+
+        LOG.info("finish backup!")
+        self._finish_backup(context, backup, volume, updates)
+
+    @utils.trace
+    def _finish_backup(self, context, backup, volume, updates):
+        volume_id = backup.volume_id
+        snapshot_id = backup.snapshot_id
+        previous_status = volume.get('previous_status', None)
+
+        # Restore the original status.
+        if snapshot_id:
+            self.db.snapshot_update(
+                context, snapshot_id,
+                {'status': fields.SnapshotStatus.AVAILABLE})
+        else:
+            self.db.volume_update(context, volume_id,
+                                  {'status': previous_status,
+                                   'previous_status': 'backing-up'})
+
+        # _run_backup method above updated the status for the backup, so it
```

Comment:
`_run_backup` doesn't exist anymore. Needs to be `_start_backup`

Reply:
done